src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -370,7 +373,14 @@ private void setState(State state) {
   }
 
   private boolean shouldRotate() {
-    return recordCounter >= flushSize;
+    if (recordCounter >= flushSize) {
+      return true;
+    } else if (rotateIntervalMs <= 0) {
+      return false;
+    } else {
+      long now = System.currentTimeMillis();
```

Comment:
This is going to end up calling `System.currentTimeMillis()` for every record, right? Because to get the record count version to work accurately, this must be called for every record. Should we lift the call out into the parent method, ensure it's only invoked once per batch and pass the value in here?